tests/mocks/search.py

```diff
@@ -162,46 +148,43 @@ def index_submit_documents(
 
             return results
 
-        if not (pointer in self._documents_by_index):
-            self._documents_by_index[pointer] = []
-
         for document in documents:
             self._documents_by_index[pointer].append(document)
 
         return []
 
+    def _pointer_set(self, revision: SearchSchemaRevision, alias: str) -> SearchPointer:
+        return SearchPointer(
+            alias=alias,
+            index=revision.name_for_index(self.base_name),
+            version=revision.version,
+        )
+
     def write_pointer_set(self, revision: SearchSchemaRevision) -> None:
         self._fail_if_necessary()
-        self._write_pointer = SearchWritePointer(self.base_name, revision.version)
+        self._write_pointer = self._pointer_set(revision, self.write_pointer_name())
 
-    def index_clear_documents(self, pointer: str):
+    def index_clear_documents(self):
         self._fail_if_necessary()
-        if pointer in self._documents_by_index:
-            self._documents_by_index[pointer] = []
+        pointer = self.write_pointer_name()
+        self._documents_by_index[pointer].clear()
 
-    def search_client(self, write=False) -> Search:
-        return self._search_client.index(
-            self.read_pointer_name() if not write else self.write_pointer_name()
-        )
+    def read_search_client(self) -> Search:
+        return self._search_client.index(self.read_pointer_name())
 
-    def search_multi_client(self, write=False) -> MultiSearch:
-        return self._multi_search_client.index(
-            self.read_pointer_name() if not write else self.write_pointer_name()
-        )
+    def read_search_multi_client(self) -> MultiSearch:
+        return self._multi_search_client.index(self.read_pointer_name())
 
-    def index_remove_document(self, pointer: str, id: int):
+    def index_remove_document(self, doc_id: int):
         self._fail_if_necessary()
-        if pointer in self._documents_by_index:
-            items = self._documents_by_index[pointer]
-            to_remove = []
-            for item in items:
-                if item.get("_id") == id:
-                    to_remove.append(item)
-            for item in to_remove:
-                items.remove(item)
-
-    def is_pointer_empty(*args):
-        return False
+        pointer = self.write_pointer_name()
+        items = self._documents_by_index[pointer]
+        to_remove = []
+        for item in items:
+            if item.get("_id") == doc_id:
+                to_remove.append(item)
```

Comment:
I see this came in from older code, but could do this with a list comprehension:
```python
 to_remove = [item for item in items if item.get("_id") == doc_id]
```